chrome/content/zotero/bindings/itembox.xml

```diff
@@ -1296,6 +1296,13 @@
 						case 'dateDue':
 						case 'accepted':
 							if (fieldName == 'date' && this.item._objectType != 'feedItem') {
+								if (valueText) {
+									var date = Zotero.Date.strToDate(valueText);
+									if (date.part == undefined) {
+										// valueText = Zotero.Date.isoToDate(date).toLocaleDateString();
+										valueText = Zotero.Date.dateToISO(date);
```

Comment:
OK, so the biggest issue here is that we can't merge this with ISO dates, at least as the default. I still believe what I said in the thread, which is that 8601 is a computer format that would be foreign to most people, and it'd also be inconsistent with the existing localized Date Added/Modified fields in the itembox and items list:

<img width="210" alt="date-columns" src="https://cloud.githubusercontent.com/assets/783082/26294348/035a8ec4-3e91-11e7-8902-d06895fec9f7.png">

(Going from, say, `Jan. 2008` to `2008-01` in the itembox would also be a significant decrease in readability for most people, which is arguably worse than the inconsistency with other items.)

So for this patch, we should output a locale date string, and we can deal with how we'd allow using ISO dates in a separate issue (and if we did that, we'd want it to affect Date Added/Modified too). Using `toLocaleDateString()` I think also means we need to additionally limit this to when values exist for `year`, `month`, and `day`.

I know your priority is ISO dates, so if you don't have time to finish this at the moment to show locale dates, I can take it from here.